lab-stuart/model/bst.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+class BST {
+  constructor(value){
+    this.value = value;
+    this.left = null;
+    this.right = null;
+  }
+
+  insert(node){
+    if(!(node instanceof BST))
+      throw new Error('Node must be a bst. ');
+    if(node.value === this.value)
+      return;
+    if(node.value < this.value){
+      if(!this.left){
+        this.left = node;
+        return;
+      }
+      this.left.insert(node);
+      return;
+    }
+    if(!this.right){
+      this.right = node;
+      return;
+    }
+    this.right.insert(node);
+  }
+
+  remove(node) {
```

Comment:
Doube check this code once you write tests for it. I think it might not be properly deleting nodes in a way that preserves the BST's properties.